p2p/pex/reactor.go

```diff
@@ -387,11 +386,17 @@ func (r *ReactorV2) handleMessage(chID p2p.ChannelID, envelope p2p.Envelope) (er
 // send a request for addresses.
 func (r *ReactorV2) processPeerUpdate(peerUpdate p2p.PeerUpdate) {
 	r.Logger.Debug("received PEX peer update", "peer", peerUpdate.NodeID, "status", peerUpdate.Status)
+
+	r.mtx.Lock()
+	defer r.mtx.Unlock()
+
 	switch peerUpdate.Status {
 	case p2p.PeerStatusUp:
-		r.availablePeers.PushBack(peerUpdate.NodeID)
+		r.availablePeers[peerUpdate.NodeID] = struct{}{}
 	case p2p.PeerStatusDown:
-		r.removePeer(peerUpdate.NodeID)
+		delete(r.availablePeers, peerUpdate.NodeID)
+		delete(r.requestsSent, peerUpdate.NodeID)
+		delete(r.lastReceivedRequests, peerUpdate.NodeID)
```

Comment:
Why the need for these map deletions? How come we didn't delete them before? The last two items may still be needed, no?

Reply:
we did before: in `removePeer` which I removed: https://github.com/tendermint/tendermint/pull/6476/files#diff-bd4b05204cd4a50b093b253670b37192196d72566069a6c55c6769745dd9d5f8L497